### PR TITLE
devo7e-256 bugfix for original switch mod

### PR DESCRIPTION
**Pull request by silpstream**
Bugfix for original extra switch mod in devo7e-256.

### CODE REVIEW

**Comment by happyharrysco1** [comment]
tested and working fine on stock 7e firmware with no switch mods, ult7e with no switchmods and ult7e with 4x3, 2x2 and 2pots

File: src/target/devo7e-256/channels.c [hunk 1]
```diff
@@ -104,38 +135,100 @@ s32 CHAN_ReadRawInput(int channel)
...
-        case INP_SWE1:     value = !(global_extra_switches & 0x0c); break;
-        case INP_SWF0:     value = global_extra_switches   & 0x01;  break;
-        case INP_SWF1:     value = !(global_extra_switches & 0x03); break;
+        case INP_SWG0:     value = global_extra_switches   & 0x04;  break;
+        case INP_SWG1:     value = !(global_extra_switches & 0x0c); break;
+        case INP_SWH0:     value = global_extra_switches   & 0x01;  break;
+        case INP_SWH1:     value = !(global_extra_switches & 0x03); break;
       }
     } else {
+      if ((~Transmitter.ignore_src & SWITCH_3x1) == SWITCH_3x1) {
```
Comment (victzh) [inline src/target/devo7e-256/channels.c]: I don't like the large amount of highly repetitive code. Can you try to make it a loop over INP_SW* ?

File: src/target/emu_devo7e-256/channels.c [hunk 2]
```diff
@@ -40,105 +73,116 @@
...
+        case INP_SWB0:  return (gui.ail_dr % 3) == 0 ? CHAN_MAX_VALUE : CHAN_MIN_VALUE;
+        case INP_SWB1:  return (gui.ail_dr % 3) == 1 ? CHAN_MAX_VALUE : CHAN_MIN_VALUE;
+        case INP_SWB2:  return (gui.ail_dr % 3) == 2 ? CHAN_MAX_VALUE : CHAN_MIN_VALUE;
+        case INP_SWG0:  return (gui.mix % 2) == 0 ? CHAN_MAX_VALUE : CHAN_MIN_VALUE;
+        case INP_SWG1:  return (gui.mix % 2) == 1 ? CHAN_MAX_VALUE : CHAN_MIN_VALUE;
+        case INP_SWH0:  return (gui.fmod % 2)   ? CHAN_MIN_VALUE : CHAN_MAX_VALUE;
+        case INP_SWH1:  return (gui.fmod % 2)   ? CHAN_MAX_VALUE : CHAN_MIN_VALUE;
+      }
+    } else {
+      if ((~Transmitter.ignore_src & SWITCH_3x1) == SWITCH_3x1) {
```
Comment (victzh) [inline src/target/emu_devo7e-256/channels.c]: And repetitive code here.

**Comment by silpstream** [comment]
Hi Victzh, I'd be glad to. Problem is, I'm not quite sure how to do that. I'm not that great a programmer. Any chance you could point me in the right direction? Thanks a lot for taking a look at this!

**Pull request merged**